### PR TITLE
dev, v2.1, v3.0: fix broken links

### PR DESCRIPTION
**Pull request by CaitinChen**
### What is changed, added or deleted? 

Fix broken links.

### Which version does your change affect? 

dev, v2.1, v3.0

### Checklist 

- [x] Add a new file to `TOC.md`
- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

@lilin90 PTAL

### CODE REVIEW

File: v2.1-legacy/op-guide/docker-compose.md [hunk 1]
```diff
@@ -160,4 +160,4 @@ docker-compose exec tispark-master /opt/spark/bin/pyspark
 docker-compose exec tispark-master /opt/spark/bin/sparkR
 ```
 
-更多关于 TiSpark 的信息，参见 [TiSpark 的详细文档](../tispark/tispark-quick-start-guide.md)。
+更多关于 TiSpark 的信息，参见 [TiSpark 的详细文档](/how-to/get-started/tispark.md)。
```
Comment (lilin90) [inline v2.1-legacy/op-guide/docker-compose.md]:
```suggestion
更多关于 TiSpark 的信息，参见 [TiSpark 的详细文档](/v2.1-legacy/how-to/get-started/tispark.md)。
```

File: v2.1-legacy/overview.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ TiDB 具备如下特性：
 
 - 高度兼容 MySQL
     
-    [大多数情况下](https://www.pingcap.com/docs-cn/sql/mysql-compatibility/)，无需修改代码即可从 MySQL 轻松迁移至 TiDB，分库分表后的 MySQL 集群亦可通过 TiDB 工具进行实时迁移。
+    [大多数情况下](/reference/mysql-compatibility.md)，无需修改代码即可从 MySQL 轻松迁移至 TiDB，分库分表后的 MySQL 集群亦可通过 TiDB 工具进行实时迁移。
```
Comment (lilin90) [inline v2.1-legacy/overview.md]:
```suggestion
    [大多数情况下](/v2.1-legacy/reference/mysql-compatibility.md)，无需修改代码即可从 MySQL 轻松迁移至 TiDB，分库分表后的 MySQL 集群亦可通过 TiDB 工具进行实时迁移。
```

File: v2.1-legacy/sql/aggregate-group-by-functions.md [hunk 3]
```diff
@@ -56,7 +56,7 @@ mysql> select a, b, sum(c) from t group by a;
 ERROR 1055 (42000): Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'b' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
 ```
 
-目前，TiDB 默认不开启 SQL 模式 [`ONLY_FULL_GROUP_BY`](../sql/mysql-compatibility.md#默认设置的区别)。
+目前，TiDB 默认不开启 SQL 模式 [`ONLY_FULL_GROUP_BY`](/reference/mysql-compatibility.md#默认设置的区别)。
```
Comment (lilin90) [inline v2.1-legacy/sql/aggregate-group-by-functions.md]:
```suggestion
目前，TiDB 默认不开启 SQL 模式 [`ONLY_FULL_GROUP_BY`](/v2.1-legacy/reference/mysql-compatibility.md#默认设置的区别)。
```

File: v2.1-legacy/sql/character-set-configuration.md [hunk 4]
```diff
@@ -5,6 +5,6 @@ category: user guide
 
 # 字符集配置
 
-目前，TiDB 只支持 `utf8` 字符集，相当于 MySQL 中的 `utf8mb4`。MySQL 5.7 的默认字符集为 `latin1`。关于 TiDB 与 MySQL 字符集的区别，在[默认设置的区别](../sql/mysql-compatibility.md#默认设置的区别)中有相关说明。
+目前，TiDB 只支持 `utf8` 字符集，相当于 MySQL 中的 `utf8mb4`。MySQL 5.7 的默认字符集为 `latin1`。关于 TiDB 与 MySQL 字符集的区别，在[默认设置的区别](/reference/mysql-compatibility.md#默认设置的区别)中有相关说明。
```
Comment (lilin90) [inline v2.1-legacy/sql/character-set-configuration.md]:
```suggestion
目前，TiDB 只支持 `utf8` 字符集，相当于 MySQL 中的 `utf8mb4`。MySQL 5.7 的默认字符集为 `latin1`。关于 TiDB 与 MySQL 字符集的区别，在[默认设置的区别](/v2.1-legacy/reference/mysql-compatibility.md#默认设置的区别)中有相关说明。
```

File: v2.1-legacy/sql/dml.md [hunk 5]
```diff
@@ -48,7 +48,7 @@ SELECT
 |`HAVING where_condition` | Having 子句与 Where 子句作用类似，Having 子句可以让过滤 GroupBy 后的各种数据，Where 子句用于在聚合前过滤记录。|
 |`ORDER BY` | OrderBy 子句用于指定结果排序顺序，可以按照列、表达式或者是 `select_expr` 列表中某个位置的字段进行排序。|
 |`LIMIT` | Limit 子句用于限制结果条数。Limit 接受一个或两个数字参数，如果只有一个参数，那么表示返回数据的最大行数；如果是两个参数，那么第一个参数表示返回数据的第一行的偏移量（第一行数据的偏移量是 0），第二个参数指定返回数据的最大条目数。|
-|`FOR UPDATE` | 对查询结果集所有数据上读锁（对于在查询条件内，但是不在结果集的数据，将不会加锁，如事务启动后由其他事务写入的数据），以监测其他事务对这些的并发修改。TiDB 使用[乐观事务模型](../sql/mysql-compatibility.md#事务)在语句执行期间不会检测锁冲突，在事务的提交阶段才会检测事务冲突，如果执行 Select For Update 期间，有其他事务修改相关的数据，那么包含 Select For Update 语句的事务会提交失败。|
+|`FOR UPDATE` | 对查询结果集所有数据上读锁（对于在查询条件内，但是不在结果集的数据，将不会加锁，如事务启动后由其他事务写入的数据），以监测其他事务对这些的并发修改。TiDB 使用[乐观事务模型](/reference/mysql-compatibility.md#事务)在语句执行期间不会检测锁冲突，在事务的提交阶段才会检测事务冲突，如果执行 Select For Update 期间，有其他事务修改相关的数据，那么包含 Select For Update 语句的事务会提交失败。|
```
Comment (lilin90) [inline v2.1-legacy/sql/dml.md]:
```suggestion
|`FOR UPDATE` | 对查询结果集所有数据上读锁（对于在查询条件内，但是不在结果集的数据，将不会加锁，如事务启动后由其他事务写入的数据），以监测其他事务对这些的并发修改。TiDB 使用[乐观事务模型](/v2.1-legacy/reference/mysql-compatibility.md#事务)在语句执行期间不会检测锁冲突，在事务的提交阶段才会检测事务冲突，如果执行 Select For Update 期间，有其他事务修改相关的数据，那么包含 Select For Update 语句的事务会提交失败。|
```

File: v2.1-legacy/tispark/tispark-quick-start-guide.md [hunk 6]
```diff
@@ -5,7 +5,7 @@ category: tispark
 
 # TiSpark 快速入门指南
 
-为了让大家快速体验 [TiSpark](../tispark/tispark-user-guide.md)，通过 TiDB-Ansible 安装的 TiDB 集群中默认已集成 Spark、TiSpark jar 包及 TiSpark sample data。
+为了让大家快速体验 [TiSpark](/reference/tispark.md)，通过 TiDB-Ansible 安装的 TiDB 集群中默认已集成 Spark、TiSpark jar 包及 TiSpark sample data。
```
Comment (lilin90) [inline v2.1-legacy/tispark/tispark-quick-start-guide.md]:
```suggestion
为了让大家快速体验 [TiSpark](/v2.1-legacy/reference/tispark.md)，通过 TiDB-Ansible 安装的 TiDB 集群中默认已集成 Spark、TiSpark jar 包及 TiSpark sample data。
```

**Comment by CaitinChen** [comment]
@lilin90 Thanks. PTAL again.

File: dev/_index.md [hunk 7]
```diff
@@ -11,7 +11,7 @@ TiDB 具备如下特性：
 
 - 高度兼容 MySQL
     
-    [大多数情况下](https://www.pingcap.com/docs-cn/sql/mysql-compatibility/)，无需修改代码即可从 MySQL 轻松迁移至 TiDB，分库分表后的 MySQL 集群亦可通过 TiDB 工具进行实时迁移。
+    [大多数情况下](/dev/reference/mysql-compatibility.md)，无需修改代码即可从 MySQL 轻松迁移至 TiDB，分库分表后的 MySQL 集群亦可通过 TiDB 工具进行实时迁移。
```
Comment (lilin90) [inline dev/_index.md]:
```suggestion
    [大多数情况下](/reference/mysql-compatibility.md)，无需修改代码即可从 MySQL 轻松迁移至 TiDB，分库分表后的 MySQL 集群亦可通过 TiDB 工具进行实时迁移。
```

**Pull request merged**